### PR TITLE
OpenID connect icon to SVG

### PR DESCRIPTION
**Pull request by hiddewie**
Part of https://github.com/openstreetmap/openstreetmap-website/issues/850

Split off from https://github.com/openstreetmap/openstreetmap-website/pull/4775

This pull request replaces PNG icons for the OpenID icons with their minimized SVG variants, to ensure screens with high pixel densities can display the OpenStreetMap website with beautiful graphics.  

Tested on http://127.0.0.1:3000/login

image

image

### CODE REVIEW

File: app/helpers/user_helper.rb [hunk 1]
```diff
@@ -74,7 +74,8 @@ def auth_button_preferred(name, provider, options = {})
       image_tag("#{name}.svg",
                 :alt => t("application.auth_providers.#{name}.alt"),
                 :class => "rounded-1 me-3",
-                :size => "24") + t("application.auth_providers.#{name}.title"),
+                :width => "24px",
+                :height => "24px") + t("application.auth_providers.#{name}.title"),
```
Comment (tomhughes) [inline app/helpers/user_helper.rb:78]: This was only just changed in b4f719a8c58c5177fe3dbc330fc90b3e0ef94c4c to make it consistent with the `auth_button` method so why are we changing it back again?

**Review by tomhughes — CHANGES_REQUESTED** [review]

**Review by gravitystorm — CHANGES_REQUESTED** [review]
From [CONTRIBUTING.md](https://github.com/openstreetmap/openstreetmap-website/blob/master/CONTRIBUTING.md)

* Avoid including "fixup" commits. If you have added a fixup commit (for example to fix a rubocop warning, or because you changed your own new code) please combine the fixup commit into the commit that introduced the problem. git rebase -i is very useful for this.
* Avoid including "merge" commits. If your PR can no longer be merged cleanly (for example, an unrelated change to Gemfile.lock on master now conflicts with your PR) then please rebase your PR onto the latest master. This allows you to fix the conflicts, while keeping the PR a straightforward list of commits. If there are no conflicts, then there is no need to rebase anything.

I know @tomhughes has fixed other pull requests already, but if you can fix these problems yourself before we review your PRs, then it makes it easier for everyone.

**Comment by hiddewie** [comment]
> but if you can fix these problems yourself before we review your PRs, then it makes it easier for everyone.

Sorry, I missed this part in the contributing guidelines. I am used to the Github feature to squash PRs into a single commit into the default branch, which uses the PR title and description as resulting commit. 

I will squash my commits into a single commits from now on.

File: app/helpers/user_helper.rb (hunk 1 above)
Comment (hiddewie) [reply app/helpers/user_helper.rb]: Ah yes, I changed it back.

**Comment by gravitystorm** [comment]
> Sorry, I missed this part in the contributing guidelines. I am used to the Github feature to squash PRs into a single commit into the default branch, which uses the PR title and description as resulting commit.

No problem, thanks for making the changes

> I will squash my commits into a single commits from now on.

If it's worth having multiple commits, then please keep using multiple commits! No need to squash unnecessarily. You'll see from many of the other PRs (e.g. https://github.com/openstreetmap/openstreetmap-website/pull/4680/commits) that we use multiple commits in order to do things like allow better `git blame`, allow us to cherry-pick or revert particular commits, allow explanations of one commit separately from the rest of them, etc etc.

**Review by tomhughes — APPROVED** [review]
Looks good to me now, thanks.

**Pull request merged**